tests/dataset_runner.py

```diff
@@ -369,20 +369,35 @@ def assert_data_browser_bundles(self):
         ).to_return_value(value=True)
 
     def _assert_data_browser_bundles(self, project_shortname):
+        # Numbers here correspond to references to files for each bundle (AKA contributions). We must wait for all files
+        # references to be indexed before we continue to cleanup in order to avoid potential races.
+        # A file is ref is a contribution if it appears in the bundle with "indexed": false.
+        # For ss2 we have 2 fastqs in the primary bundle; these 2 and 18 other contributions from the secondary
+        # For optimus we have 3 fastqs in the primary bundle; these 3 and 7 other contributions from the secondary
+        # for a total of 22 and 13 contributions respectively
+        num_contributions_by_project = {'Smart-seq2': 22, 'optimus': 13}
+        expected_bundle_uuids = set(self.primary_bundle_uuids).union(self.secondary_bundle_uuids)
+        files = self.azul_agent.get_entities_by_project('files', project_shortname)
+        bundle_uuids = {bundle['bundleUuid'] for file in files for bundle in file['bundles']}
+        project_shortnames = {project_short_name
+                              for file in files
+                              for project in file['projects']
+                              for project_short_name in project['projectShortname']}
+        num_contributions = sum(1 for f in files for b in f['bundles'])
```

Comment:
```suggestion
        num_contributions = sum(1 for f in files for b in f['bundles'] if b in expected_bundle_uuids)
```

and …

Reply:
Sorry, I should have made my thinking clear. I saw the assert for the number of bundle uuids and concluded that that makes the filtering obsolete. I see why that isn't quite the case though, now that I see your sugestion to allow for `>=` expected bundles.